### ISSUE TITLE
Explore Template search functionality doesn't return expected results

### ISSUE THREAD

**Issue by bmattb** [swissspidy/web-stories-wp#11271]
## Bug Description

As a user using the search facility in the Explore Templates section, I would expect that searching for the name of a template would return that template as a search result. It doesn't

eg: "Life" does not return the template titled "Celebrity Life Story"

Similarly, searching for text in a template fails to return the template(s) that contain that text. For example:

1. "New York" 
2. "RV"
3. "Boss"

The search functionality is not providing value to the user and is frustrating to use.

I understand that this is likely due to meta on the templates not being updated after this search was implemented.

## Expected Behaviour

(Not a bug, title was left out intentionally when first searching templates because title has nothing to do with template content. Now that we know it's useful to users, we will update to include title in searchable content). 

Acceptance criteria for this is to add the template titles to the potential matches when searching dashboard > templates. 

Look to `packages/dashboard/src/app/views/exploreTemplates/templateFilters.js` for current searchable logic. Templates are static, searching them is just updating visible template state, no api is involved.

Please also add a new e2e test for this here: packages/e2e-tests/src/specs/dashboard/templates/useTemplate.js (new file should be added)

## Additional Context

- Plugin Version:
- WordPress Version:
- Operating System:
- Browser:

**Comment by BrittanyIRL** [comment]
@BrookeGraham  just addressed this here: #10675 - so that should fix this. @felipebochehin87 could you verify that?

**Comment by BrittanyIRL** [comment]
Circling back to this because I did a poor job of reading the reported bug. 

So this is actually working as intended. I agree it's rather confusing. Right now we're not searching templates based on names because names are arbitrary and not really relevant to what the template is _for_ or _contains_. We settled on using metadata and colors (see screenshot for the celebrity life template as reported). 

This would actually be a feature, updating the template search to include titles. Easy enough if users are finding that templates aren't searchable by title to add. 


Do we want to add this in? @agingoldseco @aaskedall  can I get direction from you here? 

![Screen Shot 2022-04-15 at 7.54.16 AM.png](https://images.zenhubusercontent.com/5e9dcd9a9ff5dfdaffcdf132/fea2ff98-3ecb-43b5-b1fa-4c8d3348707f)
